extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -241,7 +374,23 @@ class LifecycleTask extends BackbeatTask {
             // NoncurrentVersionExpiration Days and send expiration if
             // rules all apply
             return this._compareRulesToList(bucketData, bucketLCRules,
-                allVersionsWithStaleDate, log, versioningStatus, done);
+                allVersionsWithStaleDate, log, versioningStatus,
+                err => {
+                    if (err) {
+                        return done(err);
+                    }
+
+                    if (data.IsTruncated) {
+                        this._ncvHeapBucketClear(bucketData.target.bucket);
+                        return done();
+                    }
+
+                    this._ncvHeapObjectsClear(
```

Comment:
Should we not clear NextMarker versions only if the listing is truncated?
Also, you might want to add a comment to ease the code review.

Is this pagination logic tested somewhere?